src/shim/crypto.rs

```diff
@@ -209,6 +209,52 @@ pub fn verify_bls_sig(
     fvm_shared_latest::crypto::signature::ops::verify_bls_sig(signature, data, &addr.into())
 }
 
+/// Returns `String` error if a delegated signature is invalid.
+pub fn verify_delegated_sig(
+    signature: &[u8],
+    data: &[u8],
+    addr: &crate::shim::address::Address,
+) -> Result<(), String> {
```

Comment:
Let's use `anyhow::Result<()>`. `String` itself should rarely be an error class.

Reply:
Yes, it would need to be refactored, and it would be a good change. From briefly looking at the calls, it shouldn't be too complex.

The functions from `fvm_shared` should also be refactored (historical trivia: those were likely copied from Forest in 2021). However, given that it's a library's public interface, they should rather use a more constrained error type enumeration. Of course, refactoring `ref-fvm` methods is out of the scope of this issue, but it might be interesting to do so in the future.

Reply:
Excellent, thanks @akaladarshi!

Reply:
Done.